### PR TITLE
Design a Sign-up and Login GUI using JavaFX

### LINKED ISSUE #3463: Design a Sign-up and Login GUI using JavaFX
### Proposed title of the article

Design a Sign-up and Login GUI using JavaFX

### Introduction paragraph (2-3 paragraphs)

JavaFX is an open-source Java framework that is used to create, develop, and deliver highly portable hardware-accelerated user interfaces.
It is used to create GUI's for Rich Internet Applications (RIA), Web, desktop, and embedded and mobile applications.
Login and Sign-up forms have a unique design which makes the site presentable and of a unique design.

In this article, you will learn how to build a JavaFX Login and Sign-up User Interface(UI).

### Key takeaways

At the end of this article, you will have gained the following knowledge:

- What is JavaFX
- JavaFX libraries
- Setting up a JavaFX project
- Designing the Login and Sign-up form
- Running the application

### Pre-requisites

The basics of the article include the following:

- Java language knowledge and use
- A Java IDE is set up in your machine.
- A stable internet connection.

### References

N/A

### Templates to use as guides

- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by blacklihgt**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
`This closes #3463

### CODE REVIEW

**Comment by john-amiscaray** [comment]
Hello @blacklihgt, I will be peer-reviewing this article. I will be posting some in-line comments/feedback for any issues I need you to fix but will also be committing smaller fixes to your branch myself. Please wait for me to commit before addressing any of the in-line comments or feedback I give you. After I commit, I'll leave a comment telling you that you can start addressing the in-line comments/general feedback. Thanks!

**Comment by john-amiscaray** [comment]
To start, you need to fix the name of your folder to match [our naming conventions](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/UPLOAD_INSTRUCTIONS.md#create-article-folder-structure), and the name you used when linking your images. I.e., I need you to change the name of the folder holding your article and images to have this name: `design-a-sign-up-and-login-gui-using-javafx`

File: content/articles/Design a Sign-up and Login GUI using JavaFX/index.md [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+The results are as shown below:
+
+![Final Product](/engineering-education/design-a-sign-up-and-login-gui-using-javafx/Final-product.png "Final Product")
+
+You can find the source code and files in [this](https://github.com/RisoriTofa/A-JavaFX-Login-and-Sign-Up-Form) repository.
+
+### Conclusion
+
+JavaFX is a very stable User Interface for developers developing computer pieces of software.
+It is not only interactive but more efficient and secure since one only uses elements they need and the User Interface files are separated from the System files hence there is no intermixing which may bring confusion.
```
Comment (john-amiscaray) [inline content/articles/Design a Sign-up and Login GUI using JavaFX/index.md:535]: > It is not only interactive but more efficient and secure since one only uses elements they need and the User Interface files are separated from the System files hence there is no intermixing which may bring confusion.

This sentence is too wordy and should be broken up into 2-3 sentences or so. I wasn't sure how to fix this without removing your original meaning. This is since I wasn't sure if I understood it fully as you intend the meaning to be. Therefore, I'm leaving this up to you.

Comment (blacklihgt) [reply content/articles/Design a Sign-up and Login GUI using JavaFX/index.md]: Done!

Comment (john-amiscaray) [reply content/articles/Design a Sign-up and Login GUI using JavaFX/index.md]: It seems there was a misunderstanding, I'm sorry if I was not clear. I wanted you to not commit any changes yet because I'm working on some minor changes to this branch that I'm going to commit. It would have been easier for me if you waited for me to commit my changes but it's too late now.

**Comment by john-amiscaray** [comment]
@blacklihgt The images should also be in the new folder you created named `design-a-sign-up-and-login-gui-using-javafx`. Also, please delete the old folder named `Design a Sign-up and Login GUI using JavaFX`.

File: content/articles/Design a Sign-up and Login GUI using JavaFX/index.md (hunk 1 above)
Comment (john-amiscaray) [reply content/articles/Design a Sign-up and Login GUI using JavaFX/index.md]: > 
> 
> Done!

It seems you edited the wrong sentence. I was talking about the second sentence in the conclusion

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by john-amiscaray** [comment]
@blacklihgt I just pushed a commit, please look over it in case of any mistakes, objections, clarification needed, etc. Please let me know of any of these in the form of a comment and we'll have it settled.

**Comment by john-amiscaray** [comment]
@blacklihgt I have a couple of general concerns with this article that I want you to clear. First, there are quite a few very wordy sentences I want you to shorten. Make sure to look out for these using [this free tool](https://hemingwayapp.com/). Although be warned that many of the sentences it says are too long are just code snippets, those don't have to be changed. Also, in the section detailing the properties and styling for different components, it would more effective to just show the code for it. Putting it in the format you did makes it harder and more daunting to read.

I'll be temporarily closing this pull request for you to make these changes among the others I've talked about earlier. This is just so that we can make more room in the queue.

Thanks for your understanding, if you have any questions, comments or concerns feel free to reach out to me on Slack.

**Comment by blacklihgt** [comment]
@john-amiscaray , I've finished passing the article via Hemingway. The problem is that I can't modify it when the PR is closed. Could you open it for the next steps...

@john-amiscaray @hectorkambow

**Comment by john-amiscaray** [comment]
@blacklihgt You still haven't addressed the other issues I've mentioned that I wanted you to fix (i.e. the folder structure and the section describing how to set the properties and styles of different components). Refer to my message on Slack for details on this. Let me know if you have questions/comments/concerns about these suggestions.

**Comment by blacklihgt** [comment]
@john-amiscaray I had problems modifying it while it was closed, but its all done as instructed. Please confirm...

I'll be waiting for your feedback :eyes:

File: content/articles/Design a Sign-up and Login GUI using JavaFX/index.md (hunk 1 above)
Comment (blacklihgt) [reply content/articles/Design a Sign-up and Login GUI using JavaFX/index.md]: Is it well?

**Comment by john-amiscaray** [comment]
@blacklihgt Looking through your commits, there are a few things I want you to take note of. You had a bit of a tendency to capitalize improper nouns when only proper nouns should be capitalized. For more info on this check out [this resource](https://www.englishgrammar101.com/module-1/nouns/lesson-2/common-and-proper-nouns). 

Also, there seems to have been a bit of a misunderstanding about the critique I had about the parts where you described the properties and styles of different components. What I was suggesting was that you could, instead of listing out the properties and styles as you did in paragraphs as list items, you could just show a picture of the properties and styles set in the editor. Sorry that that was not clear to you, I haven't used this technology so there was a bit of confusion on my part. Upon looking thinking about it further, however, maybe what you could do, which won't be as time-consuming, would be to make the properties, layouts, and styles as inner lists. So for example you could have it formatted something like this:

###### Label
- Properties: 
    - `Text`: *Login* 
    - `Font`: *SanSerif 30px*
- Style:
    - `-fx-text-fill`:  *#A9A9A9*. 
- Layouts: 
    - `Pref Width`: 90
    - `Pref Height`: 30
    -  `Layout X`: 244
    -  `Layout Y`: 60

I just feel that the way you formatted it looks a little daunting to read, i.e. it was formatted in a single paragraph like so:

>- **Label**: its properties are: `Text`: *Login*, `Font`: *SanSerif 30px*. Apply the following `Style`: `-fx-text-fill`:  *#A9A9A9*. In the layouts: `Pref Width`: 90, `Pref Height`: 30, `Layout X`: 244, `Layout Y`: 60.

Making it a list would be clearer to read.

File: content/articles/design-a-sign-up-and-login-gui-using-javafx/index.md [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+- **Label**: its properties are: `Text`: *Date of Birth*, `Font`: *SanSerif 15px*. Set the following styles: `-fx-text-fill`:  *#A9A9A9*. The following layouts: `Pref Width`: 105, `Pref Height`: 20, `Layout X`: 130, `Layout Y`: 480.
+- **DatePicker**: its properties are: `Prompt Text`: *mm/dd/yyyy*. Enable the `Editable` and `Show Week number` checkboxes. Enabling them allows one to enter data into the field without using the date picker. It also shows week numbers. The layouts are: `Layout X`: 250, and `Layout Y`: 480.
+- **Label**: its properties are: `Text`: *Gender*, `Font`: *SanSerif 15px*. Set the following `Style`: `-fx-text-fill`:  *#A9A9A9*. The following are the layouts: `Pref Width`: 90, `Pref Height`: 20, `Layout X`: 130, `Layout Y`: 515.
+- **RadioButton**: its properties are: `Text`: *Male*, `Font`: *SanSerif 14px*, check the `Selected` checkbox. In the `Toggle Group` key in `Gender`. This step will group the radio buttons of the same group; hence only one is selected per instance. The following are the layouts: `Layout X`: 300, `Layout Y`: 515.
```
Comment (john-amiscaray) [inline content/articles/design-a-sign-up-and-login-gui-using-javafx/index.md:364]: > In the `Toggle Group` key in `Gender`

This sentence doesn't make sense and I don't know what you mean here. It sounds like it was supposed to be part of the previous sentence. Please revise this.

File: content/articles/design-a-sign-up-and-login-gui-using-javafx/index.md [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+
+Run the application as before using the `Shift + F10` shortcut. The results are as shown below:
+
+![Final Product](/engineering-education/design-a-sign-up-and-login-gui-using-javafx/Final-product.png "Final Product")
+
+Find the source code and files in [this](https://github.com/RisoriTofa/A-JavaFX-Login-and-Sign-Up-Form) repository.
+
+### Conclusion
+
+In conclusion, JavaFX is a stable, efficient, and secure UI library for Java developers.
```
Comment (john-amiscaray) [inline content/articles/design-a-sign-up-and-login-gui-using-javafx/index.md:496]: I would revise this conclusion. It sounds like you are concluding/proving, based on what you did in this guide, that JavaFX is a stable, efficient, and secure UI library. Instead, this guide was to teach users about the basics of the library and how to use it to create a login and signup form. Instead, maybe start off with *In this guide, we...* and go on to list the things learned.

Comment (blacklihgt) [reply content/articles/design-a-sign-up-and-login-gui-using-javafx/index.md:496]: :+1:  Understood

File: content/articles/design-a-sign-up-and-login-gui-using-javafx/index.md (hunk 2 above)
Comment (blacklihgt) [reply content/articles/design-a-sign-up-and-login-gui-using-javafx/index.md:364]: Will check it out

**Comment by hectorkambow** [comment]
@blacklihgt 

Good afternoon and thank you for submitting your topic to the EngEd program. 
If it seems this PR is still quite a bit aways from being completed. 
As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published. 
All revisions and edits should be completed on your own forked repo (as to not build up a backlog in the queue). 
We will be closing this and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 
'
cc @john-amiscaray
